src/libs/Network.js

```diff
@@ -112,6 +122,7 @@ function processNetworkRequestQueue() {
             !request.data.doNotRetry && request.data.persist
         ));
         Onyx.set(ONYXKEYS.NETWORK_REQUEST_QUEUE, retryableRequests);
+        retryRequestsAdded = true;
```

Comment:
Why do we need to set this flag?

Reply:
@marcaaron. We have a queue of requests as `networkRequestQueue`. we take retry requests and save them in `ONYXKEYS.NETWORK_REQUEST_QUEUE` in case the user closes the tab or refreshes it. Now let's say the user's internet is back but he didn't refresh the tab so `networkRequestQueue` will continue processing all those requests. This flag records this behavior. so when this is `true` we know we have processed the requests and clear the requests.
```
 // User came back online and the request queue will resume, thus we should clear the NETWORK_REQUEST_QUEUE
    if (ActiveClientManager.isClientTheLeader() && retryRequestsAdded) {
        Onyx.set(ONYXKEYS.NETWORK_REQUEST_QUEUE, []);
    }
```    

Also `Onyx.set(ONYXKEYS.NETWORK_REQUEST_QUEUE, retryableRequests);`. We reset the queue here but what will happen if the user is sending requests from multiple tabs. We will overwrite the old requests. Do we want this to behave like this or we should merge those requests `Onyx.merge`?